### ISSUE TITLE
bug: logical cluster admin can't access both internal/external endpoints

### ISSUE THREAD

**Issue by hardys** [kcp-dev/kcp#2872]
### Describe the bug

Currently the logical cluster admin kubeconfig is used to access both the [external front-proxy endpoint](https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/tenancy/workspace/workspace_controller.go#L194..L195), and also the [shard directly via the baseURL](https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/tenancy/workspace/workspace_reconcile.go#L71..L72) - this won't work in the case where different CA certs are used for shard and front-proxy (such as in the [helm-charts](https://github.com/kcp-dev/helm-charts))

This only works in CI because we use the same serving-ca.crt for both shard and front-proxy, which is Ok for development but probably not a common choice for production deployments.

### Steps To Reproduce

1. Deploy KCP in a configuration where different CA certs are used for shard and front-proxy (such as in the [helm-charts](https://github.com/kcp-dev/helm-charts))
2. Create a workspace
3. Note that the workspace is stuck `Initializing`

The logs show that the single CA from the logical-cluster-admin kubeconfig can't access both endpoints for the two phases of scheduling (thanks to @sttts for the help figuring this out):

1. The creation of the logical cluster object. At this point the shard name hash is known (an annotation on the Workspace). We cannot go through the proxy because it does not know about the logical cluster yet.
2. the initialization phase. At this point the logical cluster exists and every access is like any other workspace access. I.e. it should go through the proxy.

### Expected Behaviour

The two phases of scheduling should complete successfully.

### Additional Context

[Discussion on slack](https://kubernetes.slack.com/archives/C021U8WSAFK/p1678103495741019)

tl;dr we agreed to add a `--external-logical-cluster-admin-kubeconfig` similar to the previous (reverted) change [adding an external admin kubeconfig](https://github.com/kcp-dev/kcp/commit/339a071a6b3fb2d4e9d3f6d72e900183a8885ca0), but limited to the permissions needed for workspace scheduling and deletion.
